invenio_drafts_resources/services/records/components/base.py

```diff
@@ -200,20 +206,34 @@ def _publish_new(self, identity, draft, record):
 
     def _publish_edit(self, identity, draft, record):
         """Action when publishing an edit to an existing record."""
-        # TODO: For published records, we should sync changes from the
-        # draft bucket to the record bucket, so that an instance could
-        # potentially allow a user to update files. For now, sync() only
-        # changes the default_preview and order
-
+        # For now, sync() only # changes the default_preview and order
```

Comment:
I think it would be simpler to simply fallow th previous code. I.e we always sync both files metadata and files from the draft to the record. I.e ``publish_edit()`` doesn't care about if changes is allowed to be made or not, all it cares about is to sync the draft to the record (if any changes were made)

``edit()`` is then fully responsible to decide if a draft is unlocked or not. This should lead to a simpler control flow.